Tests/NativeBuilderTests/ContainerBuildSnapshotterTests/DiffKeyEdgeCasesTests.swift

```diff
@@ -0,0 +1,637 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the container project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+import ContainerBuildIR
+import ContainerBuildSnapshotter
+import Foundation
+import Testing
+
+#if os(Linux)
+import Glibc
+#else
+import Darwin
+#endif
+
+@Suite
+struct DiffKeySwiftTestingSuite {
+
+    // MARK: - Utilities
+
+    struct TempDir {
```

Comment:
Should we be using the TestUtils functions instead where possible?

Reply:
good catch